src/libcore/ptr/mod.rs

```diff
@@ -1759,6 +1777,13 @@ impl<T: ?Sized> *mut T {
     /// of the returned pointer, nor can it ensure that the lifetime `'a`
     /// returned is indeed a valid lifetime for the contained data.
     ///
+    /// When calling this method, you have to ensure that if the pointer is
+    /// non-NULL, then it is properly aligned, dereferencable (for the whole
+    /// size of `T`) and points to an initialized instance of `T`. This applies
+    /// even if the result of this method is unused!
+    /// (The part about being initialized is not yet fully decided, but until
+    /// it is the only safe approach is to ensure that they are indeed initialized.)
+    ///
```

Comment:
I think there should be an equivalent of `as_ref`'s paragraph about the lifetime here (responsibility to ensure there are no other references).

Reply:
I added a paragraph about this, does that look good to you?

Reply:
👍 r=me after squashing